cloudinit/cmd/status.py

```diff
@@ -425,36 +441,70 @@ def get_status_details(
                     recoverable_errors[err_type].extend(
                         current_recoverable_errors[err_type]
                     )
-            if finished == 0 and start != 0:
-                status = UXAppStatus.RUNNING
-            event_time = max(start, finished)
-            if event_time > latest_event:
-                latest_event = event_time
-    if errors:
-        status = UXAppStatus.ERROR
-    elif status == UXAppStatus.NOT_RUN and latest_event > 0:
-        status = UXAppStatus.DONE
-    if uses_systemd() and status not in (
-        UXAppStatus.NOT_RUN,
-        UXAppStatus.DISABLED,
-    ):
-        systemd_status = _get_error_or_running_from_systemd(status, wait=wait)
-        if systemd_status:
-            status = systemd_status
+    return errors, recoverable_errors
+
 
+def get_status_details(
+    paths: Optional[Paths] = None, wait: bool = False
+) -> StatusDetails:
+    """Return a dict with status, details and errors.
+
+    @param paths: An initialized cloudinit.helpers.paths object.
+    @param wait: If user has indicated to wait for cloud-init to complete.
+
+    Values are obtained from parsing paths.run_dir/status.json.
+    """
+    condition_status = ConditionStatus.PEACHY
+    paths = paths or read_cfg_paths()
+    status_file = os.path.join(paths.run_dir, "status.json")
+    result_file = os.path.join(paths.run_dir, "result.json")
+    boot_status_code, boot_description = get_bootstatus(
+        CLOUDINIT_DISABLED_FILE, paths, wait
+    )
+    status_v1 = {}
+    if os.path.exists(status_file):
+        status_v1 = load_json(load_text_file(status_file)).get("v1", {})
+
+    datasource = get_datasource(status_v1)
+    description = get_description(status_v1, boot_description)
+
+    latest_event = get_latest_event(status_v1)
     last_update = (
         strftime("%a, %d %b %Y %H:%M:%S %z", gmtime(latest_event))
         if latest_event
         else ""
     )
 
-    if recoverable_errors:
-        status = UXAppStatusDegradedMap.get(status, status)
+    errors, recoverable_errors = get_errors(status_v1)
+    if errors:
+        condition_status = ConditionStatus.ERROR
+    elif recoverable_errors:
+        condition_status = ConditionStatus.DEGRADED
+
+    running_status = get_running_status(
+        status_file, result_file, boot_status_code, latest_event
+    )
+
+    if (
+        running_status == RunningStatus.RUNNING
+        and uses_systemd()
```

Comment:
We might eventually want to abstract `systemd_failed()` into the `distros/` so that other init systems can be better extended to check status. This isn't something we need to do today, just an idea for the future.